### ISSUE TITLE
Set ReportPage printable from code

### ISSUE THREAD

**Issue by silverhawk89** [fralx/LimeReport#277]
Hi. How can I set the printable  option of each ReportPage from code?

**Comment by fralx** [comment]
Hi!
You can use a variable for this purpose.
For example in the init script, you can write something like this
``` 
Reportpage2.printable = getVariable("SecondPage")
true
```

**Comment by silverhalk** [comment]
I don't understand your answer, I never work with init script before. Can you send me an example? There is no alternative in the way    report->dataManager()->setReportVariable() ? . It would not be cleaner and simpler if there was a way to access all the variables of the report page (Like printable,geometryLocked,pageisToc,printerName etc) in the same way that you can access the variables created in the report  with report->dataManager()->setReportVariable() ?. Only this time don't be dataManager() would be something like reportPages(). Instead of retrieving the data source wi'll be retrieving the report pages's object and instead of setting the reportVariable wi'll be using the setPageVariable() or setPageSettingsVariable to access those reportPages's settings. This would provide the possibility, for example, of having a single template to generate a report, allowing at runtime the decision to show or hide an specific page of the report without having to edit the template or create several, which becomes cumbersome in large projects having to change all the templates when editing one,  which favors a greater possibility of making mistakes. It would also be an alternative solution to  issue #139  by @liufeijin. Providing flexibility and adaptability to the report by allowing to set the printer to be used on each page at runtime and not when making the report. Can you consider to add this kind of functionality  to your program? Tell me if I can be of any help :)

**Comment by fralx** [comment]
I will think about your suggestion :) 
By the way, if you use init script you can change any aspects of your report via variables. 
It is rather simple :) https://youtu.be/Bts4Sq5CyKw

**Comment by silverhalk** [comment]
Hi. Use your suggestion and it worked, but I had some doubts. At the end of the script you put the word "true". What use does it have? When my script has multiple lines, do I not have to separate them with a punctuation mark?

**Comment by fralx** [comment]
> At the end of the script you put the word "true". What use does it have?

The init script should return true if you don't want to break the render process that why I use true as last instruction :)

> When my script has multiple lines, do I not have to separate them with a punctuation mark?

As you wish, limereport uses javascript syntax and in simple cases, you needn't separate lines with a punctuation mark.

**Comment by silverhalk** [comment]
Thanks @fralx !!   My program is working excellent due to your suggestions.

**Comment by liufeijin** [comment]
> > At the end of the script you put the word "true". What use does it have?
> 
> The init script should return true if you don't want to break the render process that why I use true as last instruction :)
> 
> > When my script has multiple lines, do I not have to separate them with a punctuation mark?
> 
> As you wish, limereport uses javascript syntax and in simple cases, you needn't separate lines with a punctuation mark.

hi fralx 
i use this  , but the page still print out 

CLKHPg9.beforeRender.connect(page9)
function page9(){
   CLKHPg9.printable=getVariable("CLKHPg9Printable");
   CLKHPg9.printable=false; // test
}
